### PR TITLE
Update KullbackLeibler.lean

### PR DESCRIPTION
**Pull request by LorenzoLuccioli**
Add some lemmas.
Refactor some proof to improve performance.
This PR depends on #17.

### CODE REVIEW

**Comment by RemyDegenne** [comment]
Merge master?

**Comment by LorenzoLuccioli** [comment]
We can merge, but I don't have permission to do it

File: TestingLowerBounds/KullbackLeibler.lean [hunk 1]
```diff
@@ -57,6 +60,12 @@ lemma kl_of_not_ac (h : ¬ μ ≪ ν) : kl μ ν = ⊤ := if_neg (not_and_of_not
 lemma kl_of_not_integrable (h : ¬ Integrable (llr μ ν) μ) : kl μ ν = ⊤ :=
   if_neg (not_and_of_not_right _ h)
 
+--This lemma is to make some proof a bit easier, since we can avoid repeating the integrability hypothesis if we have a cast to the reals. Unfortunately this cannot be used if we don't have the absolute continuity, since in that case the integral may still be finite but not zero.
+lemma kl_toReal_of_ac (h : μ ≪ ν) : (kl μ ν).toReal = ∫ a, llr μ ν a ∂μ := by
```
Comment (RemyDegenne) [inline TestingLowerBounds/KullbackLeibler.lean]:
```suggestion
/-- If `μ ≪ ν`, then `toReal` of the Kullback-Leibler divergence is equal to an integral,
without any integrability condition. Not true in general without `μ ≪ ν`, as the integral might be
finite but non-zero. -/
lemma kl_toReal_of_ac (h : μ ≪ ν) : (kl μ ν).toReal = ∫ a, llr μ ν a ∂μ := by
```

File: TestingLowerBounds/KullbackLeibler.lean [hunk 2]
```diff
@@ -561,4 +604,7 @@ lemma kl_pi_const {ι : Type*} [hι : Fintype ι] [CountablyGenerated α] [IsPro
 end Tensorization
 
 end ProbabilityTheory
--- TODO : add the chain rule, product version, then the tensorization for kl (see the one for the Renyi divergence)
+--TODO: check if the EReal are a metrizable space (I think the istance is not there, since using a lemma it says that it failed to sintethize the instance of pseudo metrizable space), if there is not, we could add it, we can metrize the EReal using the metric d(x,y) = |arctg(x)-arctg(y)|. This may be useful to apply some lemmas, for example
+--TODO: define the extended exp and log
+#check Measurable.stronglyMeasurable
+--TODO: bump mathlib, I tried to do it using `lake -R -Kenv=dev update` but it failed, giving me the error `function expected at FetchM`
```
Comment (RemyDegenne) [inline TestingLowerBounds/KullbackLeibler.lean]: Let's clean those todos. They are partly outdated and we don't need them in that file on the repository.
```suggestion

```

**Review by RemyDegenne — APPROVED** [review]

**Pull request merged**